api/v1beta2/cryostat_types.go

```diff
@@ -69,6 +69,10 @@ type CryostatSpec struct {
 	// +optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec
 	Resources *ResourceConfigList `json:"resources,omitempty"`
+	// Additional configuration options for the authorization proxy.
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Authorization Options",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:advanced"}
+	AuthorizationOptions *AuthorizationOptions `json:"authorizationOptions,omitempty"`
```

Comment:
Can the existing AuthProperties be used for this? Or just remove the AuthProperties and have this instead?

Reply:
I think it makes sense to remove `AuthProperties` - despite the similarity in naming, these two configuration objects have very different contents and meanings. This new one is (currently) just for providing the secret for htpasswd-based Basic, whereas the old one was for configuring the fine-grained role mapping with the old baked-in OAuth2 client (https://github.com/cryostatio/cryostat/blob/main/src/main/resources/io/cryostat/net/openshift/OpenShiftAuthManager.properties).

I was planning to do all the cleanup and removal of unused configuration items like this in a later PR, similar to #795, but if you'd rather I can clean that piece up in this PR since it is somewhat related.

Reply:
Separate PR is fine with me